Metric value import ui

chuenlok opened the pull request:
There are two duplicated files, `mod/wikirate/set/abstract/import.rb`, `mod/wikirate_source/set/abstract/import.rb`. I tried to `include Card::Set::Abstract::Import` in `mod/wikirate_source/set/type_plus_right/source/file/import.rb` but it is not working. I need to create `mod/wikirate_source/set/abstract/import.rb` in order to make it work.

Do I miss sth to include a abstract module from another mod?

xithan committed:
started on scored metrics

xithan committed:
Merge remote-tracking branch 'chuenlok/merge_subcards_api' into scored_metrics

xithan committed:
working on scored metrics

xithan committed:
working on rating

xithan committed:
update wikirate test db

xithan committed:
Merge remote-tracking branch 'wagn/master' into ratings

xithan committed:
add rubocop config file

xithan committed:
rating specs

xithan committed:
Gemfile

xithan committed:
remove Gemfile.lock

xithan committed:
i#

xithan committed:
fix hound config

xithan committed:
:dog:

xithan committed:
all specs passing

xithan committed:
DSL for creating metrics

xithan committed:
wagn tick

xithan committed:
:dog:

xithan committed:
wagn tick

xithan committed:
refactor metric type with template pattern

xithan committed:
add abstract calculation set

xithan committed:
add metric type sets

xithan committed:
source.rb: :dog:

xithan committed:
more :dog:

xithan committed:
update test seed data

xithan committed:
fix problem with missing MetricType constant

xithan committed:
Merge remote-tracking branch 'wagn/master' into ratings

xithan committed:
another test database tick

xithan committed:
make metric type options a cardtype
necessary to use *type plus right rules for +*metric type sets

xithan committed:
Gemfile

xithan committed:
working on formula metric

xithan committed:
working on formula metric

xithan committed:
add delayed_job script

xithan committed:
error in wikirate head

xithan committed:
Merge branch 'master' into ratings

xithan committed:
Merge branch 'overview' into ratings

xithan committed:
use rubocop-decko gem

xithan committed:
Merge remote-tracking branch 'wagn/master' into ratings

xithan committed:
minor fix

xithan committed:
Merge branch 'master' into ratings

xithan committed:
workint on new metric form

xithan committed:
cached count specs

xithan committed:
metric imports and specs

xithan committed:
add scripts

xithan committed:
Merge branch 'stages' into scored_metrics

xithan committed:
Merge remote-tracking branch 'origin/ratings' into all_metrics

	mod/wikirate_source/set/type/source/source.rb.orig

xithan committed:
working on new metric form

xithan committed:
working on metrics

xithan committed:
import card for new metric form

xithan committed:
working on new metric form

xithan committed:
working on new metric form

xithan committed:
update metric form migration

xithan committed:
add designer migration

xithan committed:
metric form wagneering

xithan committed:
working on score editor

xithan committed:
metric thumbnail

xithan committed:
Merge remote-tracking branch 'origin/all_metrics' into all_metrics

xithan committed:
working on score

xithan committed:
working on metric seed data

xithan committed:
tweak metric create API

xithan committed:
fix #researched?

xithan committed:
refactor metrics

xithan committed:
more metric seed data

xithan committed:
struggling with card_accerssor / subcards incompatibility

xithan committed:
got test metric creation working

xithan committed:
documentation for metric creation API

xithan committed:
Gemfile.lock tick

xithan committed:
Merge branch 'all_metrics' of https://github.com/xithan/wikirate into all_metrics

xithan committed:
views for categorical metrics

xithan committed:
fiddeling with coffeescript

Commit:
new type for metric value import

Commit:
update test db, refract more code

Commit:
kill hound

xithan committed:
metric.rb: improve documentation

xithan committed:
metric coffee

xithan committed:
script_metric.rb: fix path

xithan committed:
more migrations

xithan committed:
metric.js.coffee: get WikiRating formula editor working

xithan committed:
working on wikiratings

xithan committed:
testdb tick

xithan committed:
creation.rb: remove whitespace in HTML to make capybara work

xithan committed:
working on tests

Commit:
fix the duplicated abstract import

Commit:
remove unused initializer

xithan committed:
add metric dialog

xithan committed:
rename formula_input to variables

xithan committed:
fix 'add new metric' modal

xithan committed:
working on weights

xithan committed:
table.rb: tweak table API

xithan committed:
:dog:

xithan committed:
Merge remote-tracking branch 'wagn/master' into all_metrics

xithan committed:
update neighbor slot

xithan committed:
wagn tick

xithan committed:
Gemfile.lock tick

xithan committed:
remove broken codeclimate gem

xithan committed:
scores tabs

xithan committed:
beautify metric css

xithan committed:
wagn tick

xithan committed:
Merge pull request #358 from xithan/all_metrics

metrics to demo

xithan committed:
WikiRating fixes

xithan committed:
Merge pull request #361 from xithan/all_metrics

WikiRating fixes

xithan committed:
wagn tick

xithan committed:
metric view fixes

xithan committed:
import tick

xithan committed:
Merge branch 'all_metrics' of https://github.com/xithan/wikirate into all_metrics

xithan committed:
no limit

xithan committed:
Merge pull request #362 from xithan/all_metrics

remove limit in metric search

xithan committed:
bug: add metrics to new formula

xithan committed:
Merge remote-tracking branch 'origin/all_metrics' into all_metrics

xithan committed:
started more tests

xithan committed:
:dog:

xithan committed:
Merge remote-tracking branch 'wagn/demo' into demo

xithan committed:
more metric fixes

xithan committed:
import update

xithan committed:
wagn tick

xithan committed:
Merge branch 'all_metrics' into HEAD

xithan committed:
css changes

xithan committed:
move scss and coffeescript to code

xithan committed:
fixes and styling

xithan committed:
wikirate:rake#reseed_data: migrate before adding test data

xithan committed:
remove breakpoints

xithan committed:
reorganize some wikirate scripts and stylesheets

xithan committed:
Gemfile update

xithan committed:
wagn tick

xithan committed:
seed data tick

ethn commented:
I'll have to solve the inclusion issue.  The duplication is unacceptable.

ethn commented on mod/wikirate/set/abstract/import.rb:
```diff
@@ -0,0 +1,222 @@
+    '+metric' => { 'content' => metric },
+    '+company' => { 'content' => company },
+    '+value' => { 'content' => value, :type_id => PhraseID },
+    '+year' => { 'content' => "[[#{year}]]", :type_id => PointerID },
```
these types should not be necessary; aren't they in default rules?

ethn commented on mod/wikirate/set/type/metric_value_import_file.rb:
```diff
@@ -0,0 +1,187 @@
+  if errors.empty?
+    abort success: {
+      slot: {
+        success_msg: 'Import Successfully <br />'
```
br is ugly here.

ethn commented on mod/wikirate/set/abstract/import.rb:
```diff
@@ -0,0 +1,222 @@
+end
+
+def clean_corrected_company_hash
+  hash = Env.params[:corrected_company_name]
```
hack, no?

ethn commented on mod/wikirate/set/abstract/import.rb:
```diff
@@ -0,0 +1,222 @@
+end
+
+def import_csv_information
+end
```
hack without comment.

ethn commented on mod/wikirate/set/abstract/import.rb:
```diff
@@ -0,0 +1,222 @@
+  # transcode to utf8 before CSV reads it.
+  # some users upload files in non utf8 encoding.
+  # The microsoft excel may not save a CSV file in utf8 encoding
+  CSV.read(file.path, encoding: 'windows-1251:utf-8')
```
how does this explain why windows-1251 is hard coded here?

ethn commented on mod/wikirate/set/abstract/import.rb:
```diff
@@ -0,0 +1,222 @@
+
+format :html do
+  def aliases_hash
+    aliases_hash = {}
```
seems like a case for `@aliases_hash ||= begin...`

xithan committed:
fix formula, improve capistrano output

xithan commented on mod/wikirate/set/type/metric_value_import_file.rb:
```diff
@@ -0,0 +1,187 @@
+                                   corrected_company_hash
+  # add corrected name to alias
+  add_corrected_name_to_alias corrected_company_hash[correct_company_key],
+                              potential_alias_name
```
What do we want to do here? I would think that "corrected name" means the new (correct) name but what we do here is adding the old (wrong) name as alias. Looks like a bug to me but on the other hand `potential_alias_name` is explicitly set to the old name before we get the new name. So I'm not sure if it's just a naming issue.

xithan replied:
after a closer look to the code I assume it's a naming issue. The corrected name is the one that exists in the database so I guess we want to add to its alias card the other name.

ethn replied:
that sounds right.  need more code comments.

xithan commented on mod/wikirate/set/type/metric_value_import_file.rb:
```diff
@@ -0,0 +1,187 @@
+event :validate_import, :prepare_to_validate,
+      on: :update,
+      when: proc { Env.params['is_metric_import_update'] == 'true' } do
+end
```
empty event?

xithan commented on mod/wikirate_source/set/type_plus_right/source/file/import.rb:
```diff
@@ -1,237 +1,62 @@
-event :import_csv, :prepare_to_store,
+event :validate_import, :prepare_to_validate,
       on: :update,
       when: proc { Env.params['is_metric_import_update'] == 'true' } do
```
Why do we have this 'is_metric_import_update' flag? Do we update import files without intending to import something?

ethn replied:
good question.

xithan committed:
refactor metric value import

xithan committed:
Gemfile.lock update

xithan committed:
Merge branch 'metric_value_import_ui' into demo

xithan committed:
test db tick

xithan committed:
more specs and less failing specs

xithan committed:
test db tick

xithan committed:
wagn tick

xithan committed:
update Gemfile.lock

xithan committed:
Merge remote-tracking branch 'wagn/master' into all_metrics

xithan committed:
fix style

xithan committed:
metric_value_import_file.rb: redirect to import view

xithan committed:
improved error handling

xithan committed:
refactor metric value import

xithan committed:
remove restart_passenger task
 handled now by capistrano-passenger

xithan committed:
add migration for *source type right structure
[FIX]: we had only a "source type+*source type+*type plus right" structure but some of the cards we use as source types don't have cardtype source type (e.g. File)

xithan committed:
metric/creation.rb: value creation
[FEATURE]: #create_values on metric cards creates metric value cards

xithan committed:
metric value import refactoring and bug fixing

xithan committed:
wagn tick

xithan committed:
forgot a binding.pry

xithan committed:
handle redirect

chuenlok committed:
Merge pull request #1 from xithan/metric_value_import_ui

refactor metric value import

xithan committed:
Merge remote-tracking branch 'wagn/master' into merge

xithan committed:
resolve export conflicts

xithan committed:
fix Gemfile

xithan committed:
fix merge error

xithan committed:
remove metric type structure import

xithan committed:
another merge artefact

ethn committed:
clean up rubocop in gemfile

ethn committed:
Merge remote-tracking branch 'xithan/merge' into demo

ethn committed:
fixing demo specs: syntax error edition

ethn committed:
tests running

xithan committed:
more bad merge fixes

Commit:
update metric value type structure

Commit:
use render instead of nest

Commit:
Merge remote-tracking branch 'wagn/demo' into metric_value_import_ui

xithan committed:
fix related values error

xithan committed:
Merge remote-tracking branch 'wagn/demo' into demo

xithan committed:
Merge branch 'demo' into merge

xithan committed:
researched.rb: added missing args

xithan committed:
Merge remote-tracking branch 'chuenlok/demo' into merge

xithan committed:
fix bad merge again

xithan committed:
fix script for value type modal

xithan committed:
Merge pull request #374 from xithan/merge

remove metric type structure migration

Commit:
Merge remote-tracking branch 'wagn/demo' into metric_value_import_ui

Commit:
fix show wrong structure in all values list
